packages/lesswrong/components/posts/PostsPage/PostsPageWrapper.jsx

```diff
@@ -22,17 +25,28 @@ const PostsPageWrapper = ({document: post, sequenceId, version, data: {refetch},
   />
 }
 
-registerComponent("PostsPageWrapper", PostsPageWrapper,
-  [withDocument, {
-    collection: Posts,
-    queryName: 'postsSingleQuery',
-    fragmentName: 'PostsWithNavigation',
-    enableTotal: false,
-    enableCache: true,
-    ssr: true,
-    extraVariables: {
-      version: 'String',
-      sequenceId: 'String',
-    }
-  }]
-);
+const graphQLOptionsWithNavigation = {
```

Comment:
Move these up to the function call site

Reply:
I tried that and didn't like it. I am happy to move them above the component, but with their current line-count they made the function much harder to read.

Reply:
How about
```
const { document: post, refetch, loading, error } = useSingle({
  documentId,
  collection: Posts,
  queryName: 'postsSingleQuery',
  fragmentName: (version || sequenceId) ? 'PostsWithNavigation' : 'PostsPage',
  ...((version || sequenceId) && {
    extraVariableTypes: {
      version: 'String',
      sequenceId: 'String',
    },
    variables: {
      version, sequenceId
    }
  }),
  enableTotal: false,
  enableCache: true,
  ssr: true,
})
```
In general I have a strong dislike for the "put a function's named parameters into an object and then pass the whole object" pattern.

Reply:
Huh, that's weird. I think in particular in multi-parameter function that's basically the only good way of calling it. The alternative is to memorize what all the arguments to a function mean and then in the worst case pass through multiple parameter objects, which I find almost impossible to read when the objects have more than one or two keys. 

Giving them a variable name and then passing them through makes it a lot clearer that you are actually passing (in this case the graphQL options). And importantly the change you suggested would break because we would still pass through the extra variables, which is something we want to avoid in this case. 

Overall, I prefer having the logic of the component separated from the data its querying, which means I prefer to not have all this data querying logic obscure the core rendering path of my component.